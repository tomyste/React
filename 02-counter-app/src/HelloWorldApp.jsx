import PropTypes from 'prop-types';

const getMessage = () => {
    return "Nuevo Mensaje"
}

export const HelloWorldApp = ({title, subtitle}) => {
  return (
    <> 
      <h1 data-testid="test-title">{ title }</h1>    
      <h2>{ subtitle }</h2>
      <h2>{ subtitle }</h2>
      <p>{ getMessage() }</p>
    </>
  )
}

HelloWorldApp.defaultProps = {
    title: "hola default",
    subtitle: "Subtitulo default"
}

HelloWorldApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string.isRequired
}



//Los signos <> </> corresponden a un fragmento y es el remplazo del tipico div padre que se usaba en las otras versiones.