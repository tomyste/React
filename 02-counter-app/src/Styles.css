html, body {
    background-color: darkslategray;
    color: cornsilk;
    font-family: sans-serif;
    padding: 30px;
    text-align: center;
}

h1 {
    border-style: solid;
}