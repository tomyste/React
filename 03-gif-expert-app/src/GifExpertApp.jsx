import { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

const GifExpertApp = () => {

    const [categories, setCategories] = useState(['Dragon Ball Z']);

    const onAddCategory = ( newCategory) =>{
        if(categories.includes(newCategory)) return;
        
        const newArray = [newCategory, ...categories];
        setCategories(newArray);
    }

    return (
        <>
           <h1> GifExpertApp</h1>
           <AddCategory 
            onNewCategory = {onAddCategory}>
           </AddCategory>
           
            
            {   
                categories.map((category) => 
                    (<GifGrid 
                    key={ category }
                    category={ category }
                    >

                    </GifGrid>
                    ))
            }
            
            
            
           
        </>
    );
}

export default GifExpertApp;
