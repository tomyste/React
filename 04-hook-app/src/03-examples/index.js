

import * from './LoadingQuote';
import * from './Quote';