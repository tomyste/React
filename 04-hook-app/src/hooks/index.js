//Archivo de Barril para poder hacer las exportaciones en una sola linea

export * from './useCounter'
export * from './useFetch'
export * from './useForm'
export * from './useTodo'
