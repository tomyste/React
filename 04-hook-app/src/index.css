body {
  padding: 20px;
  background-color: cornsilk;
}

button {
  margin-right: 5px;
 
}





