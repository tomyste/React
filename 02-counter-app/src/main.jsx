import React from 'react';
import  ReactDOM  from 'react-dom/client';
import { HelloWorldApp } from './HelloWorldApp';

import './Styles.css'

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <HelloWorldApp title='Hola perros'/>
    </React.StrictMode>
);

