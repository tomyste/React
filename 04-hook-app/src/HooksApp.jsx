import { CounterApp} from "./01-useState/CounterApp";
import Header from "./components/HeaderApp";
import { CounterWithCustomHook} from "./01-useState/CounterWithCustomHook";
import { SimpleForm } from "./02-useEffect/SimpleForm";
import { FormWithCustomHook } from "./02-useEffect/FormWithCustomHook";
import { MultipleCustomHooks } from "./03-examples/MultipleCustomHooks";
import { FocusSceen } from "./04-useRef/FocusScreen";
import { Layout } from "./05-useLayoutEffect/Layout";
import { Memorize } from "./06-memos/Memorize";
import { MemoHook } from "./06-memos/MemoHook";
import { Padre } from "./07-tarea-memo/Padre";
import { TodoApp } from "./08-useReducer/TodoApp";
import { MainApp } from "./09-useContext/MainApp";




const HooksApp = () => {
    
    return (
        <>
            {/* <Header></Header>
            <CounterApp></CounterApp>
            <hr></hr>
            <CounterWithCustomHook></CounterWithCustomHook>
            <hr></hr>
            <SimpleForm></SimpleForm>
            <hr/>
            <FormWithCustomHook></FormWithCustomHook> */}
            {/* <MultipleCustomHooks></MultipleCustomHooks> */}
            {/* <FocusSceen></FocusSceen> */}
            {/* <Layout></Layout> */}
            {/* <Memorize></Memorize> */}
            {/* <MemoHook></MemoHook> */}
            {/* <Padre></Padre> */}
            {/* <TodoApp></TodoApp> */}
            <MainApp></MainApp>
        </>
    );
}

export default HooksApp;
