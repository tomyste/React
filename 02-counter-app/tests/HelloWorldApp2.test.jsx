import { render, screen } from '@testing-library/react'
import  {HelloWorldApp}  from  '../src/HelloWorldApp'


 describe('Pruebas en <HelloWorldApp/>', () => { 

    const title = 'Hola soy goku';
    const subTitle = 'Hola soy un subtitulo'

    test('Debe de hacer match con el snapshot', () => {
        
        const { container } = render(<HelloWorldApp title={title}/>);
        expect(container).toMatchSnapshot();
    
    });

    test('Debe mostrar el mensaje "Hola soy goku"', () => {
        
        render(<HelloWorldApp title={title} />);
        expect( screen.getByText( title ) ).toBeTruthy()

    });
    

    test('Debe de mostrar el titulo en un h1', () => {
        render(<HelloWorldApp title={title} />);
        expect(screen.getByRole('heading',{level: 1}).innerHTML).toContain(title)

    });
    

    test('Debe de mostrar el subtitulo pasado en las props', () => {
        render(<HelloWorldApp 
            title= { title }
            subtitle= { subTitle }
            />
            );

        expect(screen.getAllByText(subTitle).length).toBe(2);
    });
    
    

    


    

})

