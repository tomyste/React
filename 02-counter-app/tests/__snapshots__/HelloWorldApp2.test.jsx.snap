// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HelloWorldApp/> Debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola soy goku
  </h1>
  <h2>
    Subtitulo default
  </h2>
  <h2>
    Subtitulo default
  </h2>
  <p>
    Nuevo Mensaje
  </p>
</div>
`;
